pgxn/neon/control_plane_connector.c

```diff
@@ -785,6 +790,82 @@ HandleRename(RenameStmt *stmt)
 		return HandleRoleRename(stmt);
 }
 
+static bool
+neon_needs_fmgr_hook(Oid functionId) {
+
+	return (next_needs_fmgr_hook && (*next_needs_fmgr_hook) (functionId))
+		|| get_func_rettype(functionId) == EVENT_TRIGGEROID;
+}
+
+
+PG_FUNCTION_INFO_V1(noop);
+Datum noop(__attribute__ ((unused)) PG_FUNCTION_ARGS) { PG_RETURN_VOID();}
+
+static void
+force_noop(FmgrInfo *finfo)
+{
+    finfo->fn_addr   = (PGFunction) noop;
+    finfo->fn_oid    = InvalidOid;           /* not a known function OID anymore */
+    finfo->fn_nargs  = 0;                    /* no arguments for noop */
+    finfo->fn_strict = false;
+    finfo->fn_retset = false;
+    finfo->fn_stats  = 0;                    /* no stats collection */
+    finfo->fn_extra  = NULL;                 /* clear out old context data */
+    finfo->fn_mcxt   = CurrentMemoryContext;
+    finfo->fn_expr   = NULL;                 /* no parse tree */
+}
+
+static void
+neon_fmgr_hook(FmgrHookEventType event, FmgrInfo *flinfo, Datum *private)
```

Comment:
few quick thoughts:
* Placing this code in `control_plane_connector.c` seems a bit off. It follows established pattern though, so may be we need to rename file?
* Some comments would be nice. Even supa source had them
* We follow Supa approach here, let's mention that as well and put a link to their PR

Reply:
1. control_plane_connector is renamed to neon_ddl_handler
2. GUC is renamed to `enable_event_triggers_for_superuser` with default value `off`
3. Comments added
4. Reference to Supa PR added
5. Now even if event triggers are not enabled for superuser, they still can be fired if trigger procedure is owned by superuser. It allows to pass regression tests without explicit setting of this GUC and seems to be correct in general case: it should be safe to call procedure created by superuser with admin permissions.